File: java/com/dragon/inter/And.java
package com.dragon.inter;

import com.dragon.lexer.Token;

public class And extends Logical {
  public And(Token tok, Expr left, Expr right) {
    super(tok, left, right);
  }

  @Override
  public void jumping(StringBuilder b, int to, int from) {
    var label = from != 0 ? from : newLabel();
    left.jumping(b, 0, label);
    right.jumping(b, to, from);
    if (from == 0) {
      emitLabel(b, label);
    }
  }

}

File: cpp/inter/and.cpp
#include "cpp/inter/and.hpp"

namespace inter {
void And::jumping(std::stringstream &ss, std::uint32_t to, std::uint32_t from) {
  int label = from != 0 ? from : new_label();
  expr1()->jumping(ss, 0, label);
  expr2()->jumping(ss, to, from);
  if (!from) {
    emit_label(ss, label);
  }
}
}  // namespace inter
